core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorSidebarDirective.js

```diff
@@ -21,12 +21,34 @@ oppia.directive('simpleEditorSidebar', [function() {
     restrict: 'E',
     templateUrl: 'simpleEditor/sidebar',
     controller: [
-        '$scope', 'EditorModeService', 'SimpleEditorManagerService',
-        function($scope, EditorModeService, SimpleEditorManagerService) {
+       '$scope', 'EditorModeService', 'SimpleEditorManagerService',
+        'ScrollSyncService', 'QuestionIdService',
+        function($scope, EditorModeService, SimpleEditorManagerService,
+                 ScrollSyncService, QuestionIdService) {
       $scope.SUBFIELD_LABELS = [
         'Multiple choice', 'Correct answer', 'Hints', 'Bridge text'];
       $scope.setEditorModeToFull = EditorModeService.setModeToFull;
       $scope.questionList = SimpleEditorManagerService.getQuestionList();
+      $scope.ID_PREFIX = QuestionIdService.SIDEBAR_PREFIX;
+      $scope.getSidebarItemId = function(question, subfieldLabel) {
+        return QuestionIdService.getSidebarItemId(
+          question.getStateName(), subfieldLabel
+        );
+      };
+      $scope.scrollToField = function(question, subfieldLabel) {
+        ScrollSyncService.scrollTo(
+          QuestionIdService.getSubfieldId(
+            question.getStateName(), subfieldLabel
+          )
+        );
+      };
+      $scope.scrollToHeader = ScrollSyncService.scrollTo;
+      $scope.scrollToQuestion = function(question) {
+        ScrollSyncService.scrollTo(question.getId());
+      };
+      $scope.$on('SimpleEditorSidebarToggleCollapse', function() {
+        $scope.$apply();
```

Comment:
Interesting -- so if you $scope.$apply() at the point of $broadcast(), it doesn't work? That's a pity.

One other thing to try that might be less invasive: instead of invoking $rootScope, can you try doing a $scope.$emit() instead? That emits an event "upwards". If it's caught, then that would probably be better than starting all the way from $rootScope.